pysph/tools/geometry_stl.pyx

```diff
@@ -92,73 +98,227 @@ def _fill_triangle(triangle, h=0.1):
     return result[:, 0], result[:, 1], result[:, 2]
 
 
-def _get_neighbouring_particles(pa_src, pa_dst, radius_scale):
+def _get_stl_mesh(stl_fname, dx_triangle):
+    """Interpolates points within triangles in the stl file"""
+    m = mesh.Mesh.from_file(stl_fname)
+    x_list, y_list, z_list = [], [], []
+    for i in range(len(m.vectors)):
+        x1, y1, z1 = _fill_triangle(m.vectors[i], dx_triangle)
+        x2, y2, z2 = _get_triangle_sides(m.vectors[i], dx_triangle)
+        x_list.append(np.r_[x1, x2])
+        y_list.append(np.r_[y1, y2])
+        z_list.append(np.r_[z1, z2])
+
+    x = np.concatenate(x_list)
+    y = np.concatenate(y_list)
+    z = np.concatenate(z_list)
+    return x, y, z
+
+
+def remove_repeated_points(x, y, z, dx_triangle):
+    EPS = np.finfo(float).eps
+    pa_mesh = ParticleArray(name="mesh", x=x, y=y, z=z, h=EPS)
+    pa_grid = ParticleArray(name="grid", x=x, y=y, z=z, h=EPS)
+    pa_list = [pa_mesh, pa_grid]
+    nps = nnps.LinkedListNNPS(dim=3, particles=pa_list, radius_scale=1)
+    cdef float src_index = 1, dst_index = 0
+    nps.set_context(src_index=1, dst_index=0)
+    nbrs = UIntArray()
+    cdef list idx = []
+    for i in range(len(x)):
+        nps.get_nearest_particles(src_index, dst_index, i, nbrs)
+        neighbours = nbrs.get_npy_array()
+        idx.append(neighbours.min())
+    idx_set = set(idx)
+    idx = list(idx_set)
+    return pa_mesh.x[idx], pa_mesh.y[idx], pa_mesh.z[idx]
+
+
+def _get_stl_mesh_uniform(stl_fname, dx_triangle):
+    """Interpolates points within triangles in the stl file.
+
+        Return
+        ------
+        x, y, z : Coordinates of all points on surface of object
+        x_list, y_list, z_list: Coordinates of surface points grouped
+                                as lists corresponding to their triangle
+    """
+    m = mesh.Mesh.from_file(stl_fname)
+    x_list, y_list, z_list = [], [], []
+    for i in range(len(m.vectors)):
+        x1, y1, z1 = _fill_triangle(m.vectors[i], dx_triangle)
+        x2, y2, z2 = _get_triangle_sides(m.vectors[i], dx_triangle)
+        x_list.append(np.r_[x1, x2])
+        y_list.append(np.r_[y1, y2])
+        z_list.append(np.r_[z1, z2])
+    x = np.concatenate(x_list)
```

Comment:
This code isn't very different from the `_get_stl_mesh` above? Can the code repetition be minimized?